### ISSUE TITLE
YouTube ads are not blocked in Safari on Mac

### ISSUE THREAD

**Issue by sensi277** [pi-hole/pi-hole#542]
On Chrome and Firefox, as well as in Chrome on my Android phone, YouTube pre-roll video ads are blocked by pi-hole. In Safari, however, these ads still appear. Sidebar and in-player pop-ups are blocked just fine, but the video ads still get though. Screenshots demonstrating problem are embedded below.

This is on a Mac running macOS 10.11.5 (El Cap) and Safari 9.1.1. I made sure to disable all extensions in all three browsers before testing this.

Edit: Just realized that it is blocking most ads, most of the time. I couldn't get an ad to trigger in Firefox or Chrome, but every once in a while one will sneak though into Safari.

As an aside, video ads in the Android app also get through, but I think that may be a separate issue.

<img width="500" alt="in firefox w pi-hole" src="https://cloud.githubusercontent.com/assets/12399661/16344613/50cb8b08-39f2-11e6-939c-6847d1e41d11.png">
<img width="500" alt="in safari w pi-hole" src="https://cloud.githubusercontent.com/assets/12399661/16344614/50cbe850-39f2-11e6-979d-8512816e0fa7.png">
##

**Comment by dschaper** [comment]
Thanks for the report, I'll have to pass that on to the Mac owner of the group. @jacobsalmela

**Comment by jacobsalmela** [comment]
OK, I'll see if I can replicate this one.  I'm a bit busy with our logo process so I'll check in once that is done.

**Comment by dan019** [comment]
Unfortunately, I also see youtube ads in the IOS app. The ads are blocked in the browser. Anyone want to confirm this issue?

BTW, not a mac user so I can't test :(

**Comment by sensi277** [comment]
Can confirm, ads get through on in the iOS YouTube app on an iPad running iOS 9.3.3.

**Comment by dan019** [comment]
Just curious, how long has your iPad's DNS been pointing to the RPi?

**Comment by sensi277** [comment]
Since I set up the Pi about 3 days ago.

**Comment by dan019** [comment]
Just give it some time and see if your still getting ads in a week or so. Perhaps using the pi for longer will magically fix things, or not. If it doesn't work, then I'd be really curious how their actually getting through.

**Comment by sensi277** [comment]
Does that go for my Mac too? And is there a way to flush my DNS cache or something to see if it works without waiting?

**Comment by jacobsalmela** [comment]
I have not been able to replicate the problem.

Do you have these domains in your blacklist?  I have them in mine and do not get ads on the YouTube iOS app:
```
r4---sn-vgqs7nez.googlevideo.com
r4.sn-vgqs7nez.googlevideo.com
www.youtube-nocookie.com
youtube-ui.l.google.com
i1.ytimg.com
r17---sn-vgqsenes.googlevideo.com
r2---sn-vgqs7n7k.googlevideo.com
clients6.google.com
r1---sn-vgqsen7z.googlevideo.com
r1.sn-vgqsen7z.googlevideo.com
r20---sn-vgqs7ne7.googlevideo.com
r20.sn-vgqs7ne7.googlevideo.com
```
I have also been using my Pi-hole for quite a long time with my iPad and I do recall seeing something about caching of ads.

**Comment by PromoFaux** [comment]
@sensi277 did waiting help?

**Comment by sensi277** [comment]
No, it hasn't. YouTube ads still appear in Safari, although I haven't ever seen any in Chrome or Firefox.

And my iPad, as well as my Android phone, also still show ads in the YouTube app.

**Comment by bartreardon** [comment]
Just adding in that I'm seeing this behaviour as well in Safari and Waterfox. Chrome/Chromium are fine.

Standard pi-hole setup but using custom DNS setings if that makes a difference (i.e. I use the DNS servers I get from my ISP)

**Comment by jacobsalmela** [comment]
Discussion continues [here](https://discourse.pi-hole.net/t/youtube-10-second-black-screen/495/20)...
